src/vm-vcpu-ref/src/x86_64/interrupts.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+#![cfg(target_arch = "x86_64")]
+use kvm_bindings::kvm_lapic_state;
+
+// Yanked from byte_order
+macro_rules! generate_read_fn {
```

Comment:
Any reason for having these two macros if each is only used once?

Reply:
I think the idea was to have them as they were defined in `byte_order`. We can also remove the macro and just have it as a function.